databricks/koalas/generic.py

```diff
@@ -1279,6 +1279,8 @@ def groupby(self, by, as_index: bool = True):
             col_by = [_resolve_col(df, col_or_s) for col_or_s in by]
             return DataFrameGroupBy(df_or_s, col_by, as_index=as_index)
         if isinstance(df_or_s, Series):
+            if not isinstance(by[0], Series):
```

Comment:
You can also remove 

```python
        elif isinstance(by, tuple):
            by = [by]
```

Since `tuple` is already `Iterable`.

Reply:
i think they have slightly difference purpose,

if given `by` is `('hello', 'koalas')`

```python
elif isinstance(by, tuple):
    by = [by]  # [('hello', 'koalas')]
```

whereas

```python
elif isinstance(by, Iterable):
    by = [key if isinstance(key, (tuple, Series)) else (key,) for key in by]  # [('hello',), ('koalas',)]
```